OpenSim/Common/SmoothSegmentedFunction.cpp

```diff
@@ -163,50 +369,56 @@ _x0(x0),_x1(x1),_y0(y0),_y1(y1),_dydx0(dydx0),_dydx1(dydx1),
     }
 }
 
- SmoothSegmentedFunction::SmoothSegmentedFunction():
- _x0(SimTK::NaN),_x1(SimTK::NaN),_y0(SimTK::NaN)
-     ,_y1(SimTK::NaN),_dydx0(SimTK::NaN),_dydx1(SimTK::NaN),
-     _computeIntegral(false),_intx0x1(false),_name("NOT_YET_SET")
- {
-        _arraySplineUX.resize(0);        
-        _mXVec.resize(0);
-        _mYVec.resize(0);
-        _splineYintX = SimTK::Spline();
-        _numBezierSections = (int)SimTK::NaN;
-       
- }
-
- /*Detailed Computational Costs
- ________________________________________________________________________
-    If x is in the Bezier Curve
-                            Name     Comp.   Div.    Mult.   Add.    Assign.
+SmoothSegmentedFunction::
+  SmoothSegmentedFunction(const SimTK::Matrix& mX, const SimTK::Matrix& mY,  
+          double x0, double x1, double y0, double y1,double dydx0, double dydx1,
+          bool computeIntegral, bool intx0x1, const std::string& name):
+_x0(x0),_x1(x1),_y0(y0),_y1(y1),_dydx0(dydx0),_dydx1(dydx1),
+     _computeIntegral(computeIntegral),_intx0x1(intx0x1),_name(name)
+{
+    SmoothSegmentedFunctionParameters params(
+            SmoothSegmentedFunctionParameters(mX, mY, x0, x1, y0, y1, dydx0,
+                    dydx1, computeIntegral, intx0x1));
+
+    _smoothData = SmoothSegmentedFunctionDataLookup(params, name);
+}
+
+SmoothSegmentedFunction::SmoothSegmentedFunction()
```

Comment:
This is invalid - the implementation will segfault because this default ctor doesn't initialize `_smoothData`. You need a nullstate version of `SmoothSegmentedFunctionParameters` (i.e. you need a default ctor for `SmoothSegmentedFunctionParameters` that mimics the behavior that was observed in the old default ctor).